doc/specs/msgstake_flow.md

```diff
@@ -0,0 +1,41 @@
+# Flowchart of Output Address Edit
+
+A staked node's output address can be changed only when a transaction message
+`MsgStake` is signed by the owner of the node's current output address.  If a
+transaction is signed by the operator address, or the new output address
+specified in the transaction, it's rejected by the blockchain.  The following
+chart illustrates how a `MsgStake` is handled by the key functions.
+
+```mermaid
+flowchart LR
+  A(baseApp/runTx) --> B(x/auth/ValidateTransaction)
+	B --> C{After the upgrade?}
+	C -- yes --> D{Signed by any of:\nthe new output address\nthe operator address\nthe current output address}
+  D -- yes --> F([ok])
+  D -- no --> G([sdk:4])
+	C -- no --> E{Signed by any of:\nthe new output address\nthe operator address}
+  E -- yes --> F
+  E -- no --> G
+  A --> H(baseApp/runMsg)
+  H --> I(x/nodes/handleStake)
+  I --> J(x/nodes/keeper/ValidateValidatorStaking)
+  J --> K{Aafter the upgrade?\nChanging the output address?\nSigned by the current output address?}
```

Comment:
<img width="797" alt="Screenshot 2023-04-10 at 4 07 38 PM" src="https://user-images.githubusercontent.com/1892194/231015910-26398c27-d36e-47ca-874f-7a161610c1f0.png">

It's kind of hard to understand the flow here and what's expected. 

Again, I think multiple small TD diagrams will help clarify it